fendermint/vm/interpreter/src/fvm/exec.rs

```diff
@@ -152,21 +150,28 @@ where
     async fn deliver(
         &self,
         mut state: Self::State,
-        mut msg: Self::Message,
+        msg: Self::Message,
     ) -> anyhow::Result<(Self::State, Self::DeliverOutput)> {
         let (apply_ret, emitters, latency) = if msg.from == system::SYSTEM_ACTOR_ADDR {
             let (execution_result, latency) = measure_time(|| state.execute_implicit(msg.clone()));
             let (apply_ret, emitters) = execution_result?;
 
             (apply_ret, emitters, latency)
         } else {
-            // TODO: maybe compare the gas limits is better?
-            msg.gas_limit = msg.gas_limit.min(self.gas.available_block_gas());
+            if msg.gas_limit > state.gas_market().available_block_gas() {
+                tracing::warn!("gas limit exceed available block gas limit");
```

Comment:
Let's always try to be as informative as we can with log messages. With the previous text, an operator would've had to search for the log line to fully understand the context. This is more descriptive:

```suggestion
		// This is panic-worthy, but we suppress it to avoid liveness issues.
                tracing::warn!("[ASSERTION FAILED] message gas limit exceed available block gas limit; consensus engine is misbehaving");
```